### PR TITLE
Added logic to un ruby tests from buildsite.py

### PR DESCRIPTION
**Pull request by MatthiasWiesmann**
Added option to run ruby tests from the buildsite.py script. Cleaned up buildsite.py  and localrubycheck.sh

- `localrubycheck.sh` has a proper hash-bang.
- `buildsite.py` uses python facilities to manipulate paths.
- `buildsite.py` does not do blind execs with globs and pipes, as much as possible, processing is done in Python.
- general python cleanups.

### CODE REVIEW

File: software/util/buildsite.py [hunk 1]
```diff
@@ -1,55 +1,61 @@
...
 
-for dir in ["software/util","docs","software/gcloud","data"]:
+for dir in ['software/util','docs','software/gcloud','data']:
     if not os.path.isdir(dir):
-        print("\nRequired directory '%s' not found - Exiting\n" % dir)
+        print('\nRequired directory "%s" not found - Exiting\n' % dir)
         sys.exit(1)
 
-
+# Import strandard python libraries
```
Comment (pascalfleury) [inline software/util/buildsite.py:24]: strandard -> standard

File: software/util/buildsite.py [hunk 2]
```diff
@@ -121,36 +128,42 @@ def createMissingDir(dir):
...
+
+    print('\nPreparing GCloud files')
+    gcloud_files = glob.glob('software/gcloud/*.yaml')
+    for path in gcloud_files:
+      shutil.copy(path, gdir)
+    print('Done: copied %d files' % len(gcloud_files))
+    print('\nCreating %s from %s for version: %s' % (HANDLER_FILE, HANDLER_TEMPLATE, getVersion()))
+    with open(os.path.join(gdir, HANDLER_TEMPLATE)) as template_file:
+      with open(os.path.join(gdir, HANDLER_FILE), mode='w') as yaml_file:
+        template_data = template_file.read()
```
Comment (pascalfleury) [inline software/util/buildsite.py:159]: should this line be outside the loop? Can you read the same data multiple times from an opened file?

File: software/util/buildsite.py [hunk 3]
```diff
@@ -121,36 +128,42 @@ def createMissingDir(dir):
     if not os.path.exists(dir):
```
Comment (pascalfleury) [inline software/util/buildsite.py:128]: weird indentation wrt to the preceding lines.

File: software/util/buildsite.py [hunk 4]
```diff
@@ -121,36 +128,42 @@ def createMissingDir(dir):
...
-    os.system(cmd)
-    print("Done\n")
 
-from shutil import *
 def mycopytree(src, dst, symlinks=False, ignore=None):
-    #copes with already existing directories
+    """Copy a file-system tree, copes with already existing directories."""
     names = os.listdir(src)
     if ignore is not None:
         ignored_names = ignore(src, names)
```
Comment (pascalfleury) [inline software/util/buildsite.py:169]: 'ignoreD_names' seems unused?

File: software/util/buildsite.py [hunk 5]
```diff
@@ -277,8 +290,8 @@ def templateRender(template_path, extra_vars=None, template_instance=None):
 def checkFilePath(path):
     if not path in CHECKEDPATHS:
         CHECKEDPATHS.append(path)
-        if not path.startswith('/'):
-            path = os.getcwd() + "/" + path
+        if not os.path.isabs(path):
```
Comment (pascalfleury) [inline software/util/buildsite.py:293]: this is not necessary, os.path.join() already discards previous segments if a path element is absolute.

**Review by pascalfleury — COMMENTED** [review]
just a few comments.

File: software/util/buildsite.py (hunk 4 above)
Comment (MatthiasWiesmann) [reply software/util/buildsite.py:169]: It's used at line 177.

**Review by pascalfleury — APPROVED** [review]
Looks good, thanks!

**Review by pascalfleury — APPROVED** [review]

**Review by alex-jansen — APPROVED** [review]
nice option to include the ruby tests in local builds

**Pull request merged**